Fix/guard against missing matches

snim2 opened the pull request:
## Description of changes

- guard against missing matches, fix for log error found on NHSE
- release v2.0.1

## Testing - UPDATED

On `main` branch:

1. Spin up a local site with this plugin installed, or clone the repo into a site you already have
2. Visit http://localhost/wp-admin/plugins.php and ensure the plugin is activated
3. Visit http://localhost/wp-admin/plugins.php and either add and activate or just activate query-monitor
4. Visit http://localhost/wp-admin/edit.php?post_type=long-read and add or edit a long read to add an `h2` heading block with NO text content in the block
5. Visit the long read page in the front end, and check you see an error `Undefined array key 2`
6. Change to this branch `git checkout fix/guard-against-missing-matches`
7. Refresh the page on the front end and confirm the error has dissappeared

## Checklist
- [x] Changelog updated
- [x] If new release: major version tag to be bumped after release (see [docs](../README.md#changelog-and-versioning))

matthewpassmore commented:
@snim2 the error is being caused by an empty Heading block in the page content. Worth being more specific with the defense measure?

snim2 committed:
Add failing test for empty header content

This is something we've seen on live sites, where users
have created a heading block but not populated it with
text.

snim2 committed:
TOC generator skips empty heading blocks

Prior to this commit empty heading blocks do not appear
in the TOC but they do produce a warning in the logs,
since the reg exp matcher cannot find a tag identifier.

snim2 committed:
Release v2.0.1

snim2 commented:
Needs rebasing before merge

snim2 commented:
Rebasing
